### PR TITLE
New C++ Algorithms Cheat Sheet

### PR DESCRIPTION
**Pull request by jamie23**
**What does your Instant Answer do?**
My answer displays useful quick references for the functions in <algorithm> in C++.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Shortcut world:
http://www.cplusplus.com/reference/algorithm/

**Why did you choose this data source?**
I've seen it recommended for the original cheat sheet idea as a good source.

**Are there any other alternative (better) data sources?**
Not sure about better, I've always used this website for references when using C++.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Developers, geeks, programmers.

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
Yes,
https://duck.co/ideas/idea/6137/c-algorithms

**Are you having any problems? Do you need our help with anything?**
Would like to know if I have used the correct type of cheat sheet, I was caught between keyboard and reference as a type.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2015-09-17 at 00 03 28

-----
IA Page: https://duck.co/ia/view/cpp_algorithms_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/c++-algorithms.json [hunk 1]
```diff
@@ -0,0 +1,377 @@
+{
+    "id": "C++Algorithms_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c++-algorithms.json]: let's change this to `cpp_algorithms_cheat_sheet`. I'll updated the IA Page accordingly.

**Comment by moollaza** [comment]
@jamie23 thanks a lot for contributing, we really appreciate it! 

>Would like to know if I have used the correct type of cheat sheet, I was caught between keyboard and reference as a type.

@jamie23 I think this should use either the `code` or `reference` cheat sheet. `code` will keep the key/value beside each other, but will style it more like the terminal template looks.`

File: share/goodie/cheat_sheets/json/c++-algorithms.json [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    "name": "C++ Algorithms",
+    "description": "C++ Algorithm reference",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c++-algorithms.json]: Let's give a brief overview of the algorithms to make this a bit more descriptive.

**Comment by jamie23** [comment]
@moollaza 
Did some updates. Quick questions:
1) I've left the filename the same while still changing the id to the one specified, would you like the file name update to match the id?
2) Thats probably the best description I can provide having searched around for a bit, if you want something more in depth for each section of the cheat sheet I can put a message in the C++ slack channel and see if there is anyone who can be of more help.

**Comment by moollaza** [comment]
> I've left the filename the same while still changing the id to the one specified, would you like the file name update to match the id?

It's fine as-is. Seeing as the filename determines the default trigger, it currently allows for "c++ algorithms cheat sheet" to work.

>Thats probably the best description I can provide having searched around for a bit, if you want something more in depth for each section of the cheat sheet I can put a message in the C++ slack channel and see if there is anyone who can be of more help.

The description is great, this is good enough :+1:

**Comment by moollaza** [comment]
@jamie23 this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@jamie23 congrats! It's live: https://duckduckgo.com/?q=c%2B%2B+algorithms+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**